### PR TITLE
Allow download window to be closed on failed downloads.

### LINKED ISSUE #11843: 'Cancel' in Content download dialogue possibly broken
I noticed this on #11841, when trying to download the Firestorm assets (which failed due to lack of mirrors), only the 'Retry' button worked, the 'Cancel' button did nothing.

### PR DESCRIPTION
**Pull request by RoosterDragon**
Fixes #11843. Regression from #11745.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/Installation/DownloadPackageLogic.cs [hunk 1]
```diff
@@ -208,6 +208,7 @@ void ShowDownloadDialog()
 					if (i.Error != null)
 					{
 						onError(Download.FormatErrorMessage(i.Error));
+						cancelButton.OnClick = Ui.CloseWindow;
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Widgets/Logic/Installation/DownloadPackageLogic.cs]: Hmm... .Wouldn't it make more sense to add this line to the error handler action instead (somewhere around line 115)? Alternatively, you'd have to add this same line before line 131.

Comment (RoosterDragon) [inline OpenRA.Mods.Common/Widgets/Logic/Installation/DownloadPackageLogic.cs]: You're right - will move it to the error handler.

Comment (RoosterDragon) [inline OpenRA.Mods.Common/Widgets/Logic/Installation/DownloadPackageLogic.cs]: Done.

**Comment by obrakmann** [comment]
:+1:

**Comment by pchote** [comment]
Fix confirmed, thanks :+1:

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/eb5c2b18d8c881b079cd1075f2b5bb4cd0bb1702

**Pull request merged**